stdlib/builtins.pyi

```diff
@@ -1068,8 +1071,12 @@ class _PathLike(Protocol[_AnyStr_co]):
 
 if sys.version_info >= (3, 10):
     def aiter(__async_iterable: _SupportsAiter[_SupportsAnextT]) -> _SupportsAnextT: ...
+
+    class _SupportsSynchronousAnext(Protocol[_AwaitableT_co]):
+        def __anext__(self) -> _AwaitableT_co: ...
+
     @overload
-    async def anext(__i: SupportsAnext[_T]) -> _T: ...
+    def anext(__i: _SupportsSynchronousAnext[_AwaitableT]) -> _AwaitableT: ...
```

Comment:
here's a close python implementation of anext: https://github.com/python/cpython/blob/ea786a882b9ed4261eafabad6011bc7ef3b5bf94/Lib/test/test_asyncgen.py#L52-L80

```suggestion
    # anext is not, in fact, an async function. When default is not provided
    # anext is just a passthrough for obj.__anext__
    def anext(__i: _SupportsSynchronousAnext[_AwaitableT]) -> _AwaitableT: ...
```

maybe it's obvious now and doesn't need a comment :shrug:

Reply:
to get an idea of what I'm trying to describe, here's a close python implementation of anext: https://github.com/python/cpython/blob/ea786a882b9ed4261eafabad6011bc7ef3b5bf94/Lib/test/test_asyncgen.py#L52-L80

Reply:
Oh, that's really useful! We should definitely include a link to that in the comment!